syntax = "proto3";

package network.events;
//protoc --python_out=. network_event.proto
message NetworkEvent {
    string event_id = 1;
    int64 timestamp = 2;
    string source_ip = 3;
    string target_ip = 4;
    int32 packet_size = 5;
    int32 dest_port = 6;
    int32 src_port = 7;
    string agent_id = 8;
    float anomaly_score = 9;
    double latitude = 10;
    double longitude = 11;

    // Nuevos campos para eventos enriquecidos cuando alguien trate de usar TOR.
    string event_type = 12;
    float risk_score = 13;
    string description = 14;
}
